src/tallies/tally_filter_zernike_radial.F90

```diff
@@ -0,0 +1,203 @@
+
+    ! Get specified order
+    call get_node_value(node, "order", n)
+    this % order = n
+    this % n_bins = n/2 + 1 
+  end subroutine from_xml
+
+  subroutine get_all_bins(this, p, estimator, match)
+    class(ZernikeRadialFilter), intent(in)  :: this
+    type(Particle),      intent(in)  :: p
+    integer,             intent(in)  :: estimator
+    type(TallyFilterMatch),   intent(inout) :: match
+
+    integer :: i
+    real(8) :: x, y, r
+    real(C_DOUBLE) :: zn_rad(this % n_bins)
+
+    ! Determine normalized (r,theta) positions
+    x = p % coord(1) % xyz(1) - this % x
+    y = p % coord(1) % xyz(2) - this % y
+    r = sqrt(x*x + y*y)/this % r
+    if (r <= 1) then
+
+      ! Get moments for even order Zernike polynomial orders 0..n
+      call calc_zn_rad(this % order, r, zn_rad)
+
+      do i = 1, this % n_bins
+        call match % bins % push_back(i)
+        call match % weights % push_back(zn_rad(i))
+      end do
+    endif
+  end subroutine get_all_bins
+
+  subroutine to_statepoint(this, filter_group)
+    class(ZernikeRadialFilter), intent(in) :: this
+    integer(HID_T),      intent(in) :: filter_group
+
+    call write_dataset(filter_group, "type", "zernikeradial")
+    call write_dataset(filter_group, "n_bins", this % n_bins)
+    call write_dataset(filter_group, "order", this % order)
+    call write_dataset(filter_group, "x", this % x)
+    call write_dataset(filter_group, "y", this % y)
+    call write_dataset(filter_group, "r", this % r)
+  end subroutine to_statepoint
+
+  function text_label(this, bin) result(label)
+    class(ZernikeRadialFilter), intent(in) :: this
+    integer,             intent(in) :: bin
+    character(MAX_LINE_LEN)         :: label
+
+    label = "Zernike expansion, Z" // trim(to_str(2*bin)) // ",0"
+    ! integer :: n
+    ! integer :: first, last
+
+    ! do n = 0, this % order, -2
+      ! last = (n + 1)*(n + 2)/2
+      ! if (bin <= last) then
+      !   first = last - n
+      !   m = -n + (bin - first)*2
+        ! label = "Zernike expansion, Z" // trim(to_str(n)) // ",0"
+    !     exit
+    !   end if
+    ! end do
+  end function text_label
+
+!===============================================================================
+!                               C API FUNCTIONS
+!===============================================================================
+
+  function openmc_zernike_radial_filter_get_order(index, order) result(err) bind(C)
+    ! Get the order of an expansion filter
+    integer(C_INT32_T), value       :: index
+    integer(C_INT),     intent(out) :: order
+    integer(C_INT) :: err
+
+    err = verify_filter(index)
+    if (err == 0) then
+      select type (f => filters(index) % obj)
+      type is (ZernikeRadialFilter)
+        order = f % order
+      class default
+        err = E_INVALID_TYPE
+        call set_errmsg("Not a Zernike Radial filter.")
+      end select
+    end if
+  end function openmc_zernike_radial_filter_get_order
```

Comment:
If the radial Zernike filter was defined to `extends(ZernikeFilter)`, then the existing C API routines could be used with both a Zernike filter and a radial Zernike filter (i.e., you could get rid of this function and the ones below).